api/app/signals/apps/api/v1/views/signal.py

```diff
@@ -22,10 +21,20 @@
 from signals.auth.backend import JWTAuthBackend
 
 
-class PublicSignalViewSet(CreateModelMixin, DetailSerializerMixin, RetrieveModelMixin,
-                          PublicSignalGenericViewSet):
-    serializer_class = PublicSignalCreateSerializer
-    serializer_detail_class = PublicSignalSerializerDetail
+class PublicSignalViewSet(PublicSignalGenericViewSet):
+    def create(self, request):
+        serializer = PublicSignalCreateSerializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        signal = serializer.save()
+
+        data = PublicSignalSerializerDetail(signal, context={'request': request}).data
+        return Response(data, status=status.HTTP_201_CREATED)
+
+    def retrieve(self, request, signal_id):
```

Comment:
The `DetailSerializerMixin` should already know that this is a request to the detail endpoint and return a serializer based on the `serializer_detail_class`. So it should not be necessary to overwrite the `retrieve` function

Reply:
How important is this, I like having it repeated and explicit --- but willing to remove it.

Reply:
This is already behaviour in the current API implementation. So why duplicate it?